Sming/system/include/m_printf.h

```diff
@@ -9,16 +9,66 @@ Descr: embedded very simple version of printf with float support
 #define _M_PRINTF_
 
 #include <stdarg.h>
+#include <stdlib.h>
+#include <string.h>
 
 #ifdef __cplusplus
 extern "C" {
 #endif
 
+/** @brief callback type to output a string of data
+ *  @param param
+ *  @param str data to send
+ *  @param length number of characters to write
+ *  @retval number of characters written, which may be less than the requested size
+ *  @note data does not need to be nul terminated and may contain any 8-bit values including nul
+ */
+typedef size_t (*nputs_callback_t)(void* param, const char* str, size_t length);
+
+/** @brief set the character output routine
+ *  @param callback
+ *  @param param
+ */
+void m_setPuts(nputs_callback_t callback, void* param);
```

Comment:
@mikee47 When you have time can you check if there is any performance impact if in here we use std::function instead of good-old C callback? The std::function implementation will allow us to call not only functions and anon. functions but also object methods.

Reply:
@slaff Runs fine, new commit so you can have a look.